assets/queries/k8s/apiversion_is_deprecated/query.rego

```diff
@@ -0,0 +1,32 @@
+package Cx
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    metadata := document.metadata
+
+    recommendedVersions := {
+                            "extensions/v1beta1" : {
+                                                      "Deployment" : "apps/v1",
+                                                      "DaemonSet" : "apps/v1",
+                                                      "Ingress" : "apps/v1"
+                                                   },
+                            "apps/v1beta1" : {
+                                                      "Deployment" : "apps/v1",
+                                                      "StatefulSet" : "apps/v1",
+                                                   },
+                            "apps/v1beta2" : {
+                                                      "Deployment" : "apps/v1",
+                                                      "DaemonSet" : "apps/v1",
+                                                      "StatefulSet" : "apps/v1"
+                                                   }
+                          }
+    object.get(recommendedVersions,document.apiVersion,"undefined") != "undefined"
+    object.get(recommendedVersions[document.apiVersion],document.kind,"undefined") != "undefined"
+    result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("apiVersion=%s",[document.apiVersion]),
```

Comment:
@joelsou5a In this case I'm not using the metadata key because apiVersion comes before, so if I identify apiVersion after metadata, it will catch the next occurrence of _apiVersion_in the file, and if I put the metadata key after apiVersion, it catches the metadata key and not apiVersion.